src/app/static/src/app/components/Warning.vue

```diff
@@ -0,0 +1,164 @@
+        lineHeight: number;
+    }
+
+    interface Methods {
+        toggleShowFullBox: () => void;
+        updateDimensions: () => void;
+        headerText: (key: string) => string;
+    }
+
+    interface Computed  {
+        currentLanguage: Language;
+        renderedBoxHeight: number;
+        warningsLengthy: boolean;
+        buttonText: string;
+        maxBoxHeight: number;
+        lineStyling: Dict<string | number | undefined>;
+    }
+
+    export default Vue.extend<Data, Methods, Computed, Props>({
+        name: "Warning",
+        props: {
+            origin: String,
+            warnings: Array,
+            maxLines: {
+                default: 2,
+                required: false,
+                type: Number
+            }
+        },
+        data() {
+            return {
+                showFullBox: false,
+                fullBoxHeight: 0,
+                lineHeight: 0
+            };
+        },
+        computed: {
+            renderedBoxHeight(){
+                if (!this.warningsLengthy){
+                    return this.fullBoxHeight
+                } else {
+                    return this.showFullBox ? this.fullBoxHeight : this.maxBoxHeight
+                }
+            },
+            maxBoxHeight(){
+                return this.maxLines * this.lineHeight
+            },
+            warningsLengthy(){
+                return this.fullBoxHeight > this.maxBoxHeight
+            },
+            currentLanguage: mapStateProp<RootState, Language>(
+                null,
+                (state: RootState) => state.language
+            ),
+            buttonText(){
+                return i18next.t(this.showFullBox ? "showLess" : "showMore", { lng: this.currentLanguage });
+            },
+            // The user has the option to configure the number of lines that will be displayed. 
+            // If the height of the warnings rendered are more than this number, each warning 
+            // will be truncated to one line and the show more button will appear. If there is 
+            // only one warning message, it will be truncated after the max number of lines
+            lineStyling(){
+                if (this.warningsLengthy && !this.showFullBox){
+                    if (this.warnings.length === 1){
+                        return {
+                            height: `${this.maxBoxHeight}px`,
+                            overflow: "hidden",
+                            "display": "-webkit-box",
+                            "-webkit-line-clamp": this.maxLines,
+                            "-webkit-box-orient": "vertical"
+                        }
+                    } else {
+                        return {
+                            height: `${this.lineHeight}px`,
+                            whiteSpace: "nowrap",
+                            overflow: "hidden",
+                            textOverflow: "ellipsis"
+                        }
+                    }
```

Comment:
I reckon the second if statement can be separated from the lineStyling().

Reply:
Not quite sure what you mean. Could you elaborate?

Reply:
You can separate the highlighted snippet and call it from inside lineStyling() instead of having all the codes inside lineStyling()

Reply:
The code is very specific to this purpose and is unlikely to be reusable elsewhere so I think I'll keep it all together.